### DISCUSSION TITLE
using validations with model in a component

### DISCUSSION THREAD

**Discussion by blenke** [Megabit/Blazorise#5715] [Q&A]
I have a component based on TextEdit, I would like to use validation on this component. It uses Validations with a Model, however when doing the ValidateAll it won't work. Looks something like this:

```
<Validations @ref="ValidationsRef" Mode="ValidationMode.Manual" Model=course>
    <Validation>
        <MyTextEdit @bind-Text=course.Name />
    </Validation>
</Validations>
<Button Background="Background.Primary" @onclick=Validate>Validate</Button>
@code 
{
    Validations ValidationsRef;
    void Validate()
    {
        ValidationsRef.ValidateAll();
    }
}
```

------------------- MyTextEdit -----------
 ```
       <TextEdit @bind-Text=@Text />     

@code {
    [Parameter]
    public string Text { get; set; } = "";

    [Parameter]
    public EventCallback<string> TextChanged { get; set; }
}
```
------------ Course.cs ----------
```
    public class Course 
    {
        public int Id { get; set; }
        [MinLength(3, ErrorMessage = "Name is too short.")]
        public string Name { get; set; } = "";
    }
```
Is there a way to pass the Model to the component so that it knows what to use for the Validation ? If I use a ValidatorRule then it works just fine. When I pass the entire class as parameter and use course.Name in the component it also works but this would required a component for each field. I'm hoping i can somehow pass the class.property.

**Comment by stsrki** [comment]
Check this https://github.com/Megabit/Blazorise/discussions/1722

**Reply by blenke** [reply]
Great, that works as expected. Could you please add it to the documentation.

**Reply by stsrki** [reply]
Seems more like a blog material as it covers some of the Blazor fundamentals.

**Comment by brauerj-gc** [comment]
What if I have multiple validations I need to validate multiple components in a subcomponent?  I just successfully implemented a component with a single validation like #1722 but I'm not seeing how that works for multiple field validations in the sub-component.  

`
<EditForm Model="OrgUnitAction">
    <Validations @ref="@OrgUnitApp.NewActionValidationsRef" Mode="ValidationMode.Manual" Model="@OrgUnitAction" ValidateOnLoad="false">
   -- Validations work here --

   <SingleUnitActionForm OrgUnitAction="@OrgUnitAction" />
   </Validations>
</EditForm>
`

I tried 
`<SingleUnitActionForm OrgUnitAction="@OrgUnitAction"  OrgUnitActionExpression="() => OrgUnitAction" />`

but I wasn't sure how that should work with my use of a custom component inside of that.
SingleUnitActionForm.Razor:
`
   <MyCustomDate TValue="DateTime?" @bind-Date="@OrgUnitAction.VoteDate">
       <Feedback>
           <ValidationError/>
       </Feedback>
   </MyCustomDate>
`

Some examples of this in samples might be helpful.
